subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Output directories inside build dir
+set(src_out ${ZEPHYR_BINARY_DIR}/source/generated)
+set(include_out ${ZEPHYR_BINARY_DIR}/include/generated)
+
+# Make sure that output directory for *.c files exist
+file(MAKE_DIRECTORY ${src_out})
+
+# This file is used as the source for the parser generator
+set(cddl_file ${CMAKE_CURRENT_LIST_DIR}/modem_update.cddl)
+
+# These are the entry types needed by the source code
+set(entry_types Wrapper Sig_structure1 Segments)
+
+set(decode_c_name modem_update_decode.c)
+set(decode_h_name modem_update_decode.h)
+set(types_h_name modem_update_types.h)
+set(decode_c ${src_out}/${decode_c_name})
+set(decode_h ${include_out}/${decode_h_name})
+set(types_h ${include_out}/${types_h_name})
+set(src_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/src)
+set(include_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/include)
+set(license ${CMAKE_CURRENT_LIST_DIR}/license.txt)
+
+function(cat IN_FILE OUT_FILE)
+  file(READ ${IN_FILE} CONTENTS)
+  file(APPEND ${OUT_FILE} "${CONTENTS}")
+endfunction()
+
+foreach(target decode_c decode_h types_h)
+  list(APPEND CAT_COMMANDS COMMAND cat ${license} ${${target}} > ${${target}})
```

Comment:
Since when has `cat` become natively available on windows ?
Secondly, use of `>` and similar should be minimized if content can be handled directly.

You could do this in a cmake or python script.
In a `license.cmake`, you could do:
```
set(LICENSE "\
/*
 * Copyright (c) 2020 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 */

")

file(READ ${CMAKE_CURRENT_LIST_DIR}/${FILE} SOURCE_CONTENT)
file(WRITE ${CMAKE_CURRENT_LIST_DIR}/${FILE} ${LICENSE}${SOURCE_CONTENT})
```
and then call it using:
```
COMMAND ${CMAKE_COMMAND} -DFILE=${${target}} -P ${CMAKE_CURRENT_LIST_DIR}/license.cmake
```

This could even be improved further, by providing `-DFILES="${decode_c};${decode_h};${types_h}"` and thus remove the need for `foreach` for licence handling in this CMake file.

Reply:
since when did anyone use windows? (other than 99% of our users) :laughing: 

Anyway, thanks for the detailed proposal, will fix.

Reply:
> since when did anyone use windows?  (other than 99% of our users)

which is exactly why I imaginelly execute all CMake code in my mentally processed Windows machine.

Reply:
thinking a bit more on this.

When we generate code and commit it, we can claim copyright and decide the license.
If a NCS user generates the code, are we in principle allowed to claim Nordic copyright on that code ?